### DISCUSSION TITLE
Easy way to handle positive objective functions

### DISCUSSION THREAD

**Discussion by sambitmishra98** [meta-pytorch/botorch#1465] [Q&A] [answered]
I am trying to minimise wall-time for a simulation. I know that the value would always be positive and I must somehow enforce it on my GP model. As a novice in coding, I couldn't find an easy enough solution for this that I could implement. Some tutorials do this but I wasn't able to use those for my problem.

My first question is, what is the easiest way to constrain the objective to be above a fixed known value?

For a certain X input, If I could model the Y output wall-time as a Poission distribution or a Weibull distribution, all my problems relating to unrealistic wall-functions will be solved. Based on what I have been reading, too many things are made simple with using GP models. So just to confirm, my second question is, is it possible to easily switch the GP model to a Poisson Process model?

**Comment by saitcakmak** [comment, marked as answer]
One way to model a known positive value with GP is to model the logarithm of the value with a GP instead. So, the `log Y` can take any value from `(-inf, inf)` but the `Y = exp ( log Y )` will be restricted to `(0, inf)`. You can use similar transforms to encode greater than a known constant type constraints.

> is it possible to easily switch the GP model to a Poisson Process model?

We do not have a Poisson Process model that is available out of box in BoTorch. You would have to implement one yourself.

**Reply by sambitmishra98** [reply]
Great idea! I forgot about this trick.

**Reply by sambitmishra98** [reply]
I am trying to do `ChainedOutcomeTranform` where I combine `Log` and `Standardize` transforms together for this. @saitcakmak Could you help me out a bit?

**Reply by sambitmishra98** [reply]
Also, Log transform for noisy observations doesn't seem to be implemented yet. I have the mean and variance data for each of the observations.

**Reply by saitcakmak** [reply]
I believe that's because there isn't an analytic formula for log-transforming the variance observations. If we assume that the original observations have normally distributed observation noise, then the log-transformed observations will not have a normally distributed observation noise (which is assumed by GP).
